### PR TITLE
Introduce opt-in / out broadcast events pattern

### PR DESCRIPTION
**Pull request by jordanbyron**
Allow models to specify which events they should broadcast

### CODE REVIEW

**Comment by zaithottakath** [comment]
@wireframe what do we need to do to merge these changes and publish the gem?

File: lib/wisper/activerecord/publisher.rb [hunk 1]
```diff
@@ -15,9 +16,21 @@ module Publisher
...
-          after_commit :broadcast_destroy, on: :destroy
+          after_commit :broadcast_create, on: :create, if: -> { should_broadcast?(:create) }
+          after_commit :broadcast_update, on: :update, if: -> { should_broadcast?(:update) }
+          after_commit :broadcast_destroy, on: :destroy, if: -> { should_broadcast?(:destroy) }
+        end
+
+        class_attribute :wisper_activerecord_publisher_broadcast_events
+
+        def self.broadcast_on(*events)
+          return self.wisper_activerecord_publisher_broadcast_events = [] if events == [:none]
```
Comment (wireframe) [inline lib/wisper/activerecord/publisher.rb]: Is there another approach to effectively disabling the publisher without using `:none` as a "magic parameter" or use a constant to make it explicit?

Comment (jordanbyron) [reply lib/wisper/activerecord/publisher.rb]: @wireframe take a look at https://github.com/betterup/wisper-activerecord-publisher/pull/5/commits/1549537c8723af35ab071181f183a3b2f43497e4 and let me know what you think.

File: spec/wisper/activerecord/publisher_spec.rb [hunk 2]
```diff
@@ -27,4 +31,70 @@ class User < ActiveRecord::Base
...
+      expect(user).not_to receive(:broadcast)
+
+      user.save!
+      user.update!(name: 'Foo Bar Baz')
+      user.destroy
+    end
+  end
+
+  context 'when no broadcasts are specified' do
+    before { User.wisper_activerecord_publisher_broadcast_events = nil }
```
Comment (wireframe) [inline spec/wisper/activerecord/publisher_spec.rb:104]: is there a reason this is setting the events to `nil` vs using `disable_all_lifecycle_broadcasts!`?

File: spec/wisper/activerecord/publisher_spec.rb [hunk 3]
```diff
@@ -27,4 +31,70 @@ class User < ActiveRecord::Base
       user.destroy
     end
   end
+
+  context 'when specific broadcasts are specified' do
```
Comment (wireframe) [inline spec/wisper/activerecord/publisher_spec.rb:38]: I think there may be value in explicitly adding one more spec here.  the behavior i'd be looking to ensure is:
```
Given two models have been defined
When one model is configured with specific broadcasts
I expect the other model's broadcast events to _not_ be affected
```

The current implementation should fulfill this requirement, and this is more of an explicit test to ensure that the class_attribute does not change in the future to be reused at the Mixin level which would be undesirable behavior.

**Comment by wireframe** [comment]
great work on the edits @jordanbyron!  I left two small questions/comments and I think we're good to merge this in!  (planning for a major version bump for the new functionality).

Side note: this is a good opportunity for rolling me off of the critical path for publishing gem updates.  I'll add this to the internal roadmap for the team.

File: spec/wisper/activerecord/publisher_spec.rb (hunk 2 above)
Comment (zaithottakath) [reply spec/wisper/activerecord/publisher_spec.rb:104]: `disable_all_lifecycle_broadcasts!` sets `wisper_activerecord_publisher_broadcast_events = []` which is saying "this model has no broadcasts". setting `wisper_activerecord_publisher_broadcast_events = nil` is saying "this model has not defined any broadcasts" which means the default broadcasts are used instead.

File: spec/wisper/activerecord/publisher_spec.rb (hunk 3 above)
Comment (zaithottakath) [reply spec/wisper/activerecord/publisher_spec.rb:38]: @wireframe added!

**Comment by wireframe** [comment]
Thank you @zaithottakath and @jordanbyron !

This change has been packaged and released as part of v0.2.0.

**Pull request merged**